src/resolve/resolved-dns-question.c

```diff
@@ -418,7 +418,8 @@ int dns_question_new_service(
                 const char *type,
                 const char *domain,
                 bool with_txt,
-                bool convert_idna) {
+                bool convert_idna,
+                uint16_t record_type) {
```

Comment:
I am not convinced it makes sense to generate the PTR queries with the same function. instead, define a new function dns_question_new_service_type() or so, which generates that. After all the questions are quite different, i.e. never cover TXT, and never have a service name set...

Reply:
This was actually asked for here https://github.com/systemd/systemd/pull/22532#discussion_r992544502
 But yes, it probably makes sense to split the creation of PTR queries to a separate function

Reply:
separated creation of SRV and other queries